opentelemetry-exporter-geneva/geneva-uploader/src/payload_encoder/bond_encoder.rs

```diff
@@ -375,4 +376,51 @@ mod tests {
         let bytes = schema.as_bytes();
         assert!(!bytes.is_empty());
     }
+
+    #[test]
+    fn test_schema_buffer_capacity_estimation() {
+        // Test that the estimated capacity is adequate for various field counts
+        // and verify no reallocations occur during encoding
+
+        // Test with different field counts to validate estimation formula
+        let test_cases = vec![
+            (0, "no fields"),
+            (1, "single field"),
+            (5, "few fields"),
+            (10, "medium fields"),
+            (20, "many fields"),
+        ];
+
+        for (field_count, description) in test_cases {
+            let fields: Vec<FieldDef> = (0..field_count)
+                .map(|i| FieldDef {
+                    name: Cow::Owned(format!("field_{i}")),
+                    type_id: BondDataType::BT_STRING,
+                    field_id: i as u16 + 1,
+                })
+                .collect();
+
+            let schema = DynamicSchema::new("TestStruct", "test.namespace", fields);
+            let encoded = schema.encode().unwrap();
+
+            // Verify encoding succeeded
+            assert!(!encoded.is_empty(), "Encoding failed for {description}");
+
+            // Verify the estimated size (200 + field_count * 80) is reasonably close
+            // to the actual encoded size (allowing some tolerance)
+            let estimated_size = 200 + (field_count * 80);
+            let actual_size = encoded.len();
+
+            // The actual size should be within reasonable bounds of the estimate
+            // Allow some variance but ensure we're in the right ballpark
+            assert!(
+                actual_size <= estimated_size + 100,
```

Comment:
Why are we resorting to approximation here? I believe the only variable here is the length of the individual field names which can be computed to get the exact amount of allocation needed.